presto-spark/src/main/java/com/facebook/presto/spark/PrestoSparkExecutionFactory.java

```diff
@@ -0,0 +1,284 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.presto.Session;
+import com.facebook.presto.block.BlockEncodingManager;
+import com.facebook.presto.event.QueryMonitor;
+import com.facebook.presto.execution.QueryIdGenerator;
+import com.facebook.presto.execution.QueryInfo;
+import com.facebook.presto.execution.QueryPreparer;
+import com.facebook.presto.execution.QueryPreparer.PreparedQuery;
+import com.facebook.presto.execution.buffer.PagesSerde;
+import com.facebook.presto.execution.buffer.PagesSerdeFactory;
+import com.facebook.presto.execution.warnings.WarningCollector;
+import com.facebook.presto.operator.TaskStats;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.server.QuerySessionSupplier;
+import com.facebook.presto.server.SessionContext;
+import com.facebook.presto.spark.planner.PreparedPlan;
+import com.facebook.presto.spark.planner.SparkPlanFragmenter;
+import com.facebook.presto.spark.planner.SparkPlanPreparer;
+import com.facebook.presto.spark.planner.SparkQueryPlanner;
+import com.facebook.presto.spark.planner.SparkQueryPlanner.PlanAndUpdateType;
+import com.facebook.presto.spark.planner.SparkRddPlanner;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.SubPlan;
+import com.facebook.presto.transaction.TransactionId;
+import com.facebook.presto.transaction.TransactionInfo;
+import com.facebook.presto.transaction.TransactionManager;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slices;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.util.CollectionAccumulator;
+import scala.Some;
+import scala.Tuple2;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.airlift.concurrent.MoreFutures.getFutureValue;
+import static com.facebook.presto.SystemSessionProperties.isExchangeCompressionEnabled;
+import static com.facebook.presto.execution.buffer.PagesSerdeUtil.readSerializedPages;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class PrestoSparkExecutionFactory
```

Comment:
nit: My understanding is that `PrestoSparkExecution` corresponds to the `QueryExecution` in `presto-main`, so what about the following name:
- `PrestoSparkQueryExecution`
- `PrestoSparkQueryExecutionFactory`